pennylane/tape/tapes/reversible.py

```diff
@@ -158,9 +171,11 @@ def analytic_pd(self, idx, device, params=None, **options):
                 "reversible gradient method.".format(op.name)
             )
 
-        if self._state is None:
-            self.execute_device(params, device)
-            self._state = device._pre_rotated_state
+        # get the stored final state of the original circuit, which we start from here
+
+        final_state = self._final_state
+        # get the number of wires on the device used for the differentiation

```

Comment:
```suggestion
        # get the wires on the device used for the differentiation

```
comment was outdated...